ports/mbedTLS/azure_iot_jws_mbedtls.c

```diff
@@ -609,7 +609,7 @@ static AzureIoTResult_t prvBase64DecodeJWSHeaderAndPayload( prvJWSValidationCont
     return eAzureIoTSuccess;
 }
 
-static AzureIoTResult_t prvValidateRootKey( prvJWSValidationContext_t * pxManifestContext )
+static AzureIoTResult_t prvValidateRootKey( prvJWSValidationContext_t * pxManifestContext, AzureIoTJWS_RootKey_t * xADURootKey )
```

Comment:
This function should take an array of RootKeys: we need the `xADURootKeys` and `xADURootKeysCount`.

https://learn.microsoft.com/en-us/azure/iot-hub-device-update/device-update-security#root-keys

> Every Device Update device must contain a set of root keys. These keys are the root of trust for all of Device Update's signatures. Any signature must be chained up through one of these root keys to be considered legitimate.


We should also point to this documentation to make sure application developers are aware that the keys _will change_ over time and key management falls onto the application not the SDK.

> The set of root keys will change over time as it is proper to periodically rotate signing keys for security purposes. As a result, the Device Update agent software will need to be updated with the latest set of root keys at intervals specified by the Device Update team.

Reply:
Updated. Let me know if you think we should have it changed from the way it's currently done. But I'm going to merge this and continue pulling up changes from the embedded.